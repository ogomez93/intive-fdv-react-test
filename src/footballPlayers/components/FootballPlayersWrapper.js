import React from 'react';
import { connect } from 'react-redux';
// import { changeTesting } from '../actions';
import Table from './Table';
import FiltersWrapper from './FiltersWrapper';

const FootballPlayersWrapper = () => (
  <div>
    <FiltersWrapper />
    <Table />
  </div>
);

// const mapStateToProps = (state) => {
//   return {
//     testing: state.footballPlayers.testing
//   }
// }

// const mapDispatchToProps = (dispatch) => {
//   return {
//     onTestClick: () => {
//       dispatch(changeTesting());
//     }
//   }
// }

export default connect(
  // mapStateToProps,
  // mapDispatchToProps
)(FootballPlayersWrapper);