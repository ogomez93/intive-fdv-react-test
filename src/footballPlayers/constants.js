export const NAME = 'footballPlayers';
export const PLAYERS_API = 'https://football-players-b31f2.firebaseio.com/players.json?print=pretty';