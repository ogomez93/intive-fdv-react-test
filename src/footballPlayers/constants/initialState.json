{
  "fetching": true,
  "players": [],
  "nameFilter": "",
  "ageFilter": "",
  "positionFilter": ""
}
