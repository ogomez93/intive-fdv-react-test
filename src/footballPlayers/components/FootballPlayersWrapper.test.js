import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import FootballPlayersWrapper from './FootballPlayersWrapper';
import { initialStateFetchingTrue, initialStateFetchingFalse } from './test/constants';

const mockStore = configureStore();

const fetchingTrueStore = mockStore(initialStateFetchingTrue);
const fetchingFalseStore = mockStore(initialStateFetchingFalse);

// fetching: true

describe('Component: <FootballPlayersWrapper /> fetching players: true', () => {
  const wrapper = shallow(<FootballPlayersWrapper store={fetchingTrueStore} />);
  const component = wrapper.dive();
  const componentToJson = toJson(component);

  test('renders the component', () => {
    expect(componentToJson).toMatchSnapshot();
  });

  test('renders the title', () => {
    const title = componentToJson.children.find((el) => el.type === 'h2');
    expect(title.children[0]).toEqual('Football Player Finder');
  });

  test('renders the Filters component', () => {
    const filters = componentToJson.children.find((el) => el.type === 'Connect(Filters)');
    expect(filters).not.toBeNull();
  });

  test('application is fetching players', () => {
    const fetchingPlayers = componentToJson.children.find((el) => el.props.id === 'fetchingPlayers');
    expect(fetchingPlayers.children[0]).toEqual('Fetching players, please wait...');
  });
});

// fetching: false

describe('Component: <FootballPlayersWrapper /> fetching players: false', () => {
  const wrapper = shallow(<FootballPlayersWrapper store={fetchingFalseStore} />);
  const component = wrapper.dive();
  const componentToJson = toJson(component);

  test('renders the component', () => {
    expect(componentToJson).toMatchSnapshot();
  });

  test('renders the title', () => {
    const title = componentToJson.children.find((el) => el.type === 'h2');
    expect(title.children[0]).toEqual('Football Player Finder');
  });

  test('renders the Filters component', () => {
    const filters = componentToJson.children.find((el) => el.type === 'Connect(Filters)');
    expect(filters).not.toBeNull();
  });

  test('renders the Table component', () => {
    const table = componentToJson.children.find((el) => el.type === 'Connect(Table)');
    expect(table).not.toBeNull();
  });
});