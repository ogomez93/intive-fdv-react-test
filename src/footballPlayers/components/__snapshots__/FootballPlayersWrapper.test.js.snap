// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <FootballPlayersWrapper /> fetching players: false renders the component 1`] = `
<div>
  <h2
    className="mb-2"
  >
    Football Player Finder
  </h2>
  <Connect(Filters) />
  <Connect(Table) />
</div>
`;

exports[`Component: <FootballPlayersWrapper /> fetching players: true renders the component 1`] = `
<div>
  <h2
    className="mb-2"
  >
    Football Player Finder
  </h2>
  <Connect(Filters) />
  <div
    className="text-center"
    id="fetchingPlayers"
  >
    Fetching players, please wait...
    <div
      className="mx-auto my-12 loader"
    />
  </div>
</div>
`;
